console/program/src/data_types/plaintext_type/size_in_bits.rs

```diff
@@ -0,0 +1,145 @@
+
+                // Account for the plaintext variant bits.
+                let mut total = 2usize;
+                // Account for the number of members in the struct.
+                total = total.checked_add(8).ok_or(anyhow!("`size_in_bits` overflowed"))?;
+                // Add up the sizes of each member.
+                for (identifier, member_type) in struct_.members() {
+                    // Account for the size of the identifier.
+                    total = total.checked_add(8).ok_or(anyhow!("`size_in_bits` overflowed"))?;
+                    // Account for the identifier.
+                    total = total
+                        .checked_add(identifier.size_in_bits() as usize)
+                        .ok_or(anyhow!("`size_in_bits` overflowed"))?;
+                    // Account for the size of the member
+                    total = total.checked_add(16).ok_or(anyhow!("`size_in_bits` overflowed"))?;
+                    // Account for the member itself.
+                    let member_size = member_type.plaintext_size_in_bits_internal(get_struct, depth + 1)?;
+                    total = total.checked_add(member_size).ok_or(anyhow!("`size_in_bits` overflowed"))?;
+                }
+
+                Ok(total)
+            }
+            PlaintextType::Array(array_type) => {
+                // Account for the plaintext variant bits.
+                let mut total = 2usize;
+                // Account for the size of the array length.
+                total = total.checked_add(32).ok_or(anyhow!("`size_in_bits` overflowed"))?;
+                // Get the size of the element type.
+                let element_size =
+                    array_type.next_element_type().plaintext_size_in_bits_internal(get_struct, depth + 1)?;
+                // Get the total size of an element.
+                let element_total = 16usize.checked_add(element_size).ok_or(anyhow!("`size_in_bits` overflowed"))?;
+                // Multiply by the length of the array, ensuring no overflow occurs.
+                total = total
+                    .checked_add(
+                        element_total
+                            .checked_mul(**array_type.length() as usize)
+                            .ok_or(anyhow!("`size_in_bits` overflowed"))?,
+                    )
+                    .ok_or(anyhow!("`size_in_bits` overflowed"))?;
+
+                Ok(total)
+            }
+        }
+    }
+
+    /// Returns the number of raw bits of a plaintext type.
+    pub fn plaintext_size_in_raw_bits<F>(&self, get_struct: &F) -> Result<usize>
+    where
+        F: Fn(&Identifier<N>) -> Result<StructType<N>>,
+    {
+        self.plaintext_size_in_raw_bits_internal(get_struct, 0)
+    }
+
+    // A helper function to determine the number of raw bits of a plaintext type, while tracking the depth of the data.
+    fn plaintext_size_in_raw_bits_internal<F>(&self, get_struct: &F, depth: usize) -> Result<usize>
```

Comment:
Could you maybe add some unit tests for this function (and the one above) to confirm their correctness?